Impl vector= in srfi 133

Duolei-Wang opened the pull request:
I implement the vector= elt in srfi-133 (https://srfi.schemers.org/srfi-133/srfi-133.html#vector-eq).

The implement uses a inner compare two vectors method, and checks the type of DotList input.

When the input has a length less than 2 vectors, it return #t.

Otherwise, it compare the vectors use the compare2vec method one by one.

da-liii commented on goldfish/srfi/srfi-133.scm:
```diff
@@ -21,14 +21,40 @@
+; When the input is null or just one vector, return #t.
+; Else, it will compare the vectors in the argument list one by one.
+;     If all the vectors are equal under the elt's comparison, return #t, else #f.
+(define (vector= elt . rest)
```
use define* to simplify the code

da-liii replied:
elt is short for element, we should use elt=? here

da-liii commented on goldfish/srfi/srfi-133.scm:
```diff
@@ -21,14 +21,40 @@
+          (let loop ((vec1 (car rest))
+                     (vec2 (car (cdr rest)))
+                     (vrest (cdr (cdr rest))))
+               (if (compare2vecs elt vec1 vec2)
```
`(if 1 #t #f)` is `#t`

How about make sure the result of compare2vecs is boolean?

da-liii commented on goldfish/srfi/srfi-133.scm:
```diff
@@ -21,14 +21,40 @@
+                     (if (not (cmp (vec1 ilhs) (vec2 irhs))) #f
+                         (loop (+ 1 ilhs) (+ 1 irhs) imax))))))))
+  (if (not (procedure? elt)) (error 'type-error "elt should be a procedure")
+      (if (or(null? rest) (= 1 (length rest))) #t
```
nit: add a space after or

da-liii commented on goldfish/srfi/srfi-133.scm:
```diff
@@ -21,14 +21,40 @@
+               (if (compare2vecs elt vec1 vec2)
+                   (if (null? vrest) #t
+                       (loop vec2 (car vrest) (cdr vrest)))
+                   #f)))))
```
append an empty line here

da-liii reviewed (approved):
LGTM, but you may resolve the comments to improve the impl

LinMulikas commented on goldfish/srfi/srfi-133.scm:
```diff
@@ -21,14 +21,40 @@
+  (define compare2vecs
+    (typed-lambda ((cmp procedure?) (vec1 vector?) (vec2 vector?))
+      (let*  ((len1 (vector-length vec1))
+              (len2 (vector-length vec2)))
```
Actually, when len1 not eq len2, it already return "#f", I need not to get a "max". I will revise its name to clarify it as the same length.
```suggestion
              (len2 (vector-length vec2)))
```

Duolei-Wang commented:
I've improved the vector=, the improve they're as follows:
1. Name revise: elt to elt=?, imax to len
2. Update annotation: I deleted some unnecessary comments in source code, and add some literal description to the vector= instead of repeating it's action on the input.
3. Added a check for whether the result of compare2vecs is boolean.

Please check it again, thanks~ :)
